lib/bolt/node/ssh.rb

```diff
@@ -167,14 +176,20 @@ def make_wrapper_stringio(task_path, stdin)
 SCRIPT
     end
 
+    def make_executable(path)
+      result = execute("chmod u+x '#{path}'")
+      if result.exit_code != 0
+        raise FileError.new("Could not make file '#{path}' executable: #{result.stderr.string}", 'CHMOD_ERROR')
+      end
+    end
+
     def with_task_wrapper(remote_task, dir, stdin)
       wrapper = make_wrapper_stringio(remote_task, stdin)
       command = "#{dir}/wrapper.sh"
-      _upload(wrapper, command).then do
-        execute("chmod u+x '#{command}'")
-      end.then do
-        yield command
-      end
+      # TODO: update with_remote script to support this
```

Comment:
I'm not sure what this TODO means.

Reply:
There are too many similar functions here. We should be able to refactor them into fewer more like the winrm transport.